feature-libs/checkout/b2b/components/checkout-payment-type/checkout-payment-type.component.ts

```diff
@@ -42,22 +42,51 @@ export class CheckoutPaymentTypeComponent {
   paymentTypes$: Observable<PaymentType[]> =
     this.checkoutPaymentTypeFacade.getPaymentTypes();
 
-  typeSelected$: Observable<PaymentType> = this.checkoutPaymentTypeFacade
-    .getSelectedPaymentTypeState()
-    .pipe(
+  typeSelected$: Observable<PaymentType> = combineLatest([
+    this.checkoutPaymentTypeFacade.getSelectedPaymentTypeState().pipe(
       filter((state) => !state.loading),
-      map((state) => state.data),
-      filter(isNotUndefined),
-      distinctUntilChanged(),
-      tap((selected) => {
-        this.typeSelected = selected?.code;
-        this.checkoutStepService.resetSteps();
-        this.checkoutStepService.disableEnableStep(
-          CheckoutStepType.PAYMENT_DETAILS,
-          selected?.code === B2BPaymentTypeEnum.ACCOUNT_PAYMENT
-        );
-      })
-    );
+      map((state) => state.data)
+    ),
+    this.paymentTypes$,
+  ]).pipe(
+    map(([prevSelected, availablePaymentTypes]) => {
+      if (prevSelected) {
+        if (
+          availablePaymentTypes.find((availablePaymentType) => {
+            return availablePaymentType.code === prevSelected.code;
+          })
+        ) {
+          return prevSelected;
+        }
+      }
+      if (availablePaymentTypes.length !== 0) {
+        this.busy$.next(true);
+        this.checkoutPaymentTypeFacade
+          .setPaymentType(
+            availablePaymentTypes[0].code as string,
```

Comment:
Do we need casting `availablePaymentTypes[0].code` value to the `string`?

Reply:
Yes, In payment type interface code attribute is defined as string | undefined and in setPaymentType method input must be of type string.